### PR TITLE
bring zope.conf into the build

### PR DESCRIPTION
**Pull request by mikeAdamss**
## What is this

Currently the `zope.conf` (and the database credentials) are being injected into the container during deployment via k8s, so we're ending up with a build that's not very representative of live, it's also a pain to run locally as critical configuration is missing.

This PR moves the `zope.conf `out of the IAC and into the build and does some (not as clean as I'd like) build arg trickery so that the `zope.conf` is populated correctly per-environment and can be populated correctly when running locally against a postgres instance running via docker.

Have included how to run a local docker plone and local docker postgres from a cloudsql export in the README. Have not touched on accessing postgres local with a non containerised plone as that'll be something else again.

This isn't perfect but its probably better, principle goal is getting app config out of k8s.

## Additional Info

My preference would have been env vars rather than the hacky `sed` approach (so one build can serve multiple envs) but plone seems to have opinions. Would like to come back to this in time.

**Important** -  when or if we're happy with this approach it can only be merged in tandem with IAC cloud build changes to provide the required build args for staging as well as IAC changes to not inject `zope.conf` via k8s anymore.

This doesn''t touch on tests as (I think) they're running against a local file store plone. This doesn't resolve that but doesn't make it worse.

## How to test

Pull repo and try procedure as outlined in the README. If it works and we're happy with this the cloud build and IAC changes will  need to be a separate task.

### CODE REVIEW

**Comment by mikeAdamss** [comment]
just occurred to me we might not want to merge this until prod IAC is in 😩 as we'll have to update the old k8s configs for production as well and I don't know if we want to risk it. 70% sure it'd be fine, don't know if 70% is worth it.

**Review by eldeal — CHANGES_REQUESTED** [review]
Couple of documentation updates and typo fixes is all. Very helpful overall!!

File: plone-5/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Plone
+
+Plone requires the following env vars to run and connect to the postgres backend.
+
+| Env var | Description | Deployed Value |
+| --- | ---- | --- |
+| POSTGRES_DB | The database name used in postgres | Same as environmen, i.e sandbox/staging/production |
+| POSTGRES_USER | The user plone connect to postgres with | `plone`|
+| POSTGRES_PASSWORD | The password plone connect to postgres with | A large random string generated by terraform IAC |
+| POSTGRES_HOST | The IP address of the database | An internal GCP IP address in the appropriate VPC, named `<env>-db-internal` |
```
[the remark this answers is not in the document]
Comment (mikeAdamss) [reply plone-5/README.md:10]: no you're right, I'll change it in the IAC and update this doc when I do.

File: plone-5/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+| Env var | Description |
+| --- | ---- |
+| POSTGRES_USER | Should be `plone`|
+| POSTGRES_DB | Should be `local` |
+| POSTGRES_PASSWORD | Up to you, pick something sensible for local devlopment. |
+
+Then run postgres with docker.
+
+```
+docker run --name plone-postgres -d -p 5432:5432 postgres:14
```
[the remark this answers is not in the document]
Comment (mikeAdamss) [reply plone-5/README.md:33]: will update to a more explicit approach, it'll probably be easier for future folks to grep what's going on without the env vars anyway

File: plone-5/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikeAdamss) [reply plone-5/README.md]: for now I've removed the reference to the ip address name as I don't wan't this PR blocked waiting on the next round of IAC changes. I will change the naming convention though.

**Comment by mikeAdamss** [comment]
pr comments addressed.

it might not be where it lives in the future, but for now I've added a brief additional comment with the command to connect a local volto the the local plone setup described in the README.

**Review by eldeal — APPROVED** [review]

**Pull request merged**